kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -133,6 +139,31 @@ public UpdateControl<VirtualKafkaCluster> reconcile(VirtualKafkaCluster cluster,
         return reconciliationResult;
     }
 
+    private static void appendSecretsFromCertificateRefs(Context<VirtualKafkaCluster> context, VirtualKafkaCluster updatedCluster,
```

Comment:
This is the actual tactical fix, the VKC reconciler shouldn't really need to look into TLS let alone know about two types of thing within it.